### PR TITLE
docs(messaging): add token management info to server integration doc

### PR DESCRIPTION
**Pull request by aruhn1**
Added additional section about handling tokens while logging out users, and about tokens being associated with the app installation instance. As discussed in the discussions https://github.com/invertase/react-native-firebase/discussions/5423

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [ ] Yes
- My change supports the following platforms;
  - [ ] `Android`
  - [ ] `iOS`
- My change includes tests;
  - [ ] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [ ] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [ ] No

### Test Plan

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

File: docs/messaging/server-integration.md [hunk 1]
```diff
@@ -163,6 +163,20 @@ function App() {
 Your application code can then handle messages as you see fit; updating local cache, displaying a [notification](/messaging/notifications)
 or updating UI. The possibilities are endless!
 
+## Signing out users 
+Firebase Cloud Messaging tokens are associated with the instance of the installed app. So reinstalling the app and logging back in with the same user will generate a fresh token.
```
Comment (mikehardy) [inline docs/messaging/server-integration.md:167]: Logging in with the same user will not generate a fresh token, that would be a contradiction of the first (correct) sentence here where you explain that it is tied to the installation, not the user

```suggestion
Firebase Cloud Messaging tokens are associated with the instance of the installed app. By default, only token expiration or uninstalling/reinstalling the app will generate a fresh token.
```

File: docs/messaging/server-integration.md [hunk 2]
```diff
@@ -163,6 +163,20 @@ function App() {
 Your application code can then handle messages as you see fit; updating local cache, displaying a [notification](/messaging/notifications)
 or updating UI. The possibilities are endless!
 
+## Signing out users 
+Firebase Cloud Messaging tokens are associated with the instance of the installed app. So reinstalling the app and logging back in with the same user will generate a fresh token. 
+
+Similarly, when one user logs out and another logs in on the same app on the same device, the same FCM token will be used for both users. So if the use case involves users logging in and out on the app, the tokens should be cycled in concert.
```
Comment (mikehardy) [inline docs/messaging/server-integration.md:169]:
```suggestion
This means that by default, if your app has users and you allow them to log out and log in on the same app on the same device, the same FCM token will be used for both users. Usually this is not what you want, so you must take care to cycle the FCM token at the same time you handle user logout/login.
```

File: docs/messaging/server-integration.md [hunk 3]
```diff
@@ -163,6 +163,20 @@ function App() {
...
+
+Similarly, when one user logs out and another logs in on the same app on the same device, the same FCM token will be used for both users. So if the use case involves users logging in and out on the app, the tokens should be cycled in concert. 
+
+To log a user out, these two methods should be called back-to-back, waiting for promise resolution of course
+
+https://rnfirebase.io/reference/messaging#deleteToken
+
+https://rnfirebase.io/reference/messaging#getToken
+
+Do note that when a token is deleted by calling the `deleteToken` method, it is gone forever.
```
Comment (mikehardy) [inline docs/messaging/server-integration.md:177]:
```suggestion
How and when you invalidate a token and generate a new one will be specific to your project, but a common pattern is to delete the FCM token during logout and update your back end to remove ti, then to fetch the FCM token during login and update your back end systems to associate the new token with the logged in user.

https://rnfirebase.io/reference/messaging#deleteToken
https://rnfirebase.io/reference/messaging#getToken

Note that when a token is deleted by calling the `deleteToken` method, it is immediately and permanently invalid. 
```

**Review by mikehardy — CHANGES_REQUESTED** [review]
Thanks for posting these! I like the structure and location here. Everyone loves to offer wording suggestions and I'm no different ;-), I was a bit iffy on my exact suggestions yesterday though and after a rest and re-think, this is what I came up with. What do you think?

**Comment by aruhn1** [comment]
> Thanks for posting these! I like the structure and location here. Everyone loves to offer wording suggestions and I'm no different ;-), I was a bit iffy on my exact suggestions yesterday though and after a rest and re-think, this is what I came up with. What do you think?

I think the documentation is highly readable because people like you put so much effort into making it that way.

**Review by mikehardy — APPROVED** [review]

**Comment by mikehardy** [comment]
boom! Thanks for these. Got a separate little lint thing at the same time in #5431 so the docs are better and infrastructure made a little forward progress too. Good stuff. Cheers!

**Pull request merged**